src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -39,10 +40,15 @@ internal sealed partial class RemoteWorkspace : Workspace
         /// </summary>
         private volatile Tuple<Checksum, Solution>? _lastRequestedSolutionWithChecksum;
 
+        /// <summary>
+        /// The last partial solution snapshot corresponding to a particular project-cone requested by a service.
+        /// </summary>
+        private readonly ConcurrentDictionary<ProjectId, Tuple<Checksum, Solution>> _lastRequestedProjectIdToSolutionWithChecksum = new();
```

Comment:
>Tuple [](http://example.com/codeflow?start=57&length=5)

`StrongBox<(Checksum checksum, Solution solution)>` would be beter to avoid Item1, Item2

Reply:
ah, nice.